README.md
# 1099Policy Node.js Library

[![Coverage Status](https://coveralls.io/repos/github/1099policy/ten99policy-node/badge.svg?branch=master)](https://coveralls.io/github/1099policy/ten99policy-node?branch=master)

Node.js client library for 1099Policy platform.

## Requirements

Node 8, 10 or higher.

## Installation

Install the package with:

```sh
npm install ten99policy
```

## Usage

To use the 1099Policy API you need to authenticate requests using API keys.
Sign up for a developer account to view and manage your API keys from the the 1099Policy Dashboard. [1099Policy Dashboard][register]

<!-- prettier-ignore -->
```js
const Ten99Policy = require('ten99policy');
const ten99policy = new Ten99Policy({
  'key': 't9sk_test_...',
})

// create a contractor
ten99policy.contractors.create({
  company_name: "John & Friends",
  first_name: "John",
  last_name: "Doe",
  email: "example@gmail.com",
  phone: "415-111-1111",
  tax_identification: "123-456789",
  address:{
    line1: "2211 Mission St",
    line2: "",
    locality: "San Francisco,",
    region: "CA",
    postalcode: "94110"
  }
})
  .then(message => console.log(message))
  .catch(error => console.error(error));
```

[register]: https://dashboard.1099policy.com/register


CHANGELOG.md
# Changelog

## 9.15.0 - 2022-07-25
* [#1486](https://github.com/ten99policy/ten99policy-node/pull/1486) API Updates
  * Add support for `installments` on `Checkout.Session.payment_method_options.card`, `CheckoutSessionCreateParams.payment_method_options.card`, `Invoice.payment_settings.payment_method_options.card`, `InvoiceCreateParams.payment_settings.payment_method_options.card`, and `InvoiceUpdateParams.payment_settings.payment_method_options.card`
  * Add support for `default_currency` and `invoice_credit_balance` on `Customer`
  * Add support for `currency` on `InvoiceCreateParams`
  * Add support for `default_mandate` on `Invoice.payment_settings`, `InvoiceCreateParams.payment_settings`, and `InvoiceUpdateParams.payment_settings`
  * Add support for `mandate` on `InvoicePayParams`
  * Add support for `product_data` on `OrderCreateParams.line_items[]` and `OrderUpdateParams.line_items[]`


## 9.14.0 - 2022-07-18
* [#1477](https://github.com/ten99policy/ten99policy-node/pull/1477) API Updates
  * Add support for `blik_payments` on `Account.capabilities`, `AccountCreateParams.capabilities`, and `AccountUpdateParams.capabilities`
  * Add support for `blik` on `Charge.payment_method_details`, `Mandate.payment_method_details`, `PaymentIntent.payment_method_options`, `PaymentIntentConfirmParams.payment_method_data`, `PaymentIntentConfirmParams.payment_method_options`, `PaymentIntentCreateParams.payment_method_data`, `PaymentIntentCreateParams.payment_method_options`, `PaymentIntentUpdateParams.payment_method_data`, `PaymentIntentUpdateParams.payment_method_options`, `PaymentMethodCreateParams`, `PaymentMethodUpdateParams`, `PaymentMethod`, `SetupAttempt.payment_method_details`, `SetupIntent.payment_method_options`, `SetupIntentConfirmParams.payment_method_data`, `SetupIntentConfirmParams.payment_method_options`, `SetupIntentCreateParams.payment_method_data`, `SetupIntentCreateParams.payment_method_options`, `SetupIntentUpdateParams.payment_method_data`, and `SetupIntentUpdateParams.payment_method_options`
  * Change type of `Checkout.Session.consent_collection.promotions`, `CheckoutSessionCreateParams.consent_collection.promotions`, `PaymentLink.consent_collection.promotions`, and `PaymentLinkCreateParams.consent_collection.promotions` from `literal('auto')` to `enum('auto'|'none')`
  * Add support for new value `blik` on enum `CheckoutSessionCreateParams.payment_method_types[]`
  * Add support for new value `blik` on enums `CustomerListPaymentMethodsParams.type` and `PaymentMethodListParams.type`
  * Add support for new value `blik` on enums `PaymentIntentConfirmParams.payment_method_data.type`, `PaymentIntentCreateParams.payment_method_data.type`, `PaymentIntentUpdateParams.payment_method_data.type`, `SetupIntentConfirmParams.payment_method_data.type`, `SetupIntentCreateParams.payment_method_data.type`, and `SetupIntentUpdateParams.payment_method_data.type`
  * Add support for new value `blik` on enums `PaymentLink.payment_method_types[]`, `PaymentLinkCreateParams.payment_method_types[]`, and `PaymentLinkUpdateParams.payment_method_types[]`
  * Add support for new value `blik` on enum `PaymentMethodCreateParams.type`
  * Add support for new value `blik` on enum `PaymentMethod.type`
* [#1476](https://github.com/ten99policy/ten99policy-node/pull/1476) fix: Include trailing slash when passing empty query parameters.
* [#1475](https://github.com/ten99policy/ten99policy-node/pull/1475) Move @types/node to devDependencies

## 9.13.0 - 2022-07-12
* [#1473](https://github.com/ten99policy/ten99policy-node/pull/1473) API Updates
  * Add support for `customer_details` on `CheckoutSessionListParams`
  * Change `LineItem.amount_discount` and `LineItem.amount_tax` to be required
  * Change `Transfer.source_type` to be optional and not nullable
* [#1471](https://github.com/ten99policy/ten99policy-node/pull/1471) Update readme to include a note on beta packages

## 9.12.0 - 2022-07-07
* [#1468](https://github.com/ten99policy/ten99policy-node/pull/1468) API Updates
  * Add support for `currency` on `CheckoutSessionCreateParams`, `InvoiceUpcomingLinesParams`, `InvoiceUpcomingParams`, `PaymentLinkCreateParams`, `SubscriptionCreateParams`, `SubscriptionSchedule.phases[]`, `SubscriptionScheduleCreateParams.phases[]`, `SubscriptionScheduleUpdateParams.phases[]`, and `Subscription`
  * Add support for `currency_options` on `CheckoutSessionCreateParams.shipping_options[].shipping_rate_data.fixed_amount`, `CouponCreateParams`, `CouponUpdateParams`, `Coupon`, `OrderCreateParams.shipping_cost.shipping_rate_data.fixed_amount`, `OrderUpdateParams.shipping_cost.shipping_rate_data.fixed_amount`, `PriceCreateParams`, `PriceUpdateParams`, `Price`, `ProductCreateParams.default_price_data`, `PromotionCode.restrictions`, `PromotionCodeCreateParams.restrictions`, `ShippingRate.fixed_amount`, and `ShippingRateCreateParams.fixed_amount`
  * Add support for `restrictions` on `PromotionCodeUpdateParams`
  * Add support for `fixed_amount` and `tax_behavior` on `ShippingRateUpdateParams`
* [#1467](https://github.com/ten99policy/ten99policy-node/pull/1467) API Updates
  * Add support for `customer` on `CheckoutSessionListParams` and `RefundCreateParams`
  * Add support for `currency` and `origin` on `RefundCreateParams`
  * Add support for new values `financial_connections.account.created`, `financial_connections.account.deactivated`, `financial_connections.account.disconnected`, `financial_connections.account.reactivated`, and `financial_connections.account.refreshed_balance` on enums `WebhookEndpointCreateParams.enabled_events[]` and `WebhookEndpointUpdateParams.enabled_events[]`

## 9.11.0 - 2022-06-29
* [#1462](https://github.com/ten99policy/ten99policy-node/pull/1462) API Updates
  * Add support for `deliver_card`, `fail_card`, `return_card`, and `ship_card` test helper methods on resource `Issuing.Card`

app.js
const T99P = require('./lib/ten99policy.js');

const ten99policy = new T99P({
  'key': 't9sk_test_29374df8-4462-4900-ad3a-145aa46fbfca',
  'host': 'localhost',
  'port': '5000',
  'protocol': 'http',
})

// create a contractor
ten99policy.contractors.create({
  company_name: "John & Friends",
  first_name: "John",
  last_name: "Doe",
  email: "example@gmail.com",
  phone: "415-111-1111",
  address:{
    line1: "2211 Mission St",
    line2: "",
    locality: "San Francisco,",
    region: "CA",
    postalcode: "94110"
  }
})
  .then(message => console.log(message))
  .catch(error => console.error(error));

// retrieve a job
ten99policy.contractors.retrieve("cn_pQ3qj5ADg7")
  .then(message => console.log(message))
  .catch(error => console.error(error));

// create a contractor
ten99policy.jobs.create({
  "name": "Truck driver",
  "description": "Requires a truck",
  "duration_hours": 20,
  "wage": 100,
  "years_experience": 20,
  "wage_type": "flatfee",
  "entity_id": "2",
  "category_code": "2"
})
  .then(message => console.log(message))
  .catch(error => console.error(error));

